src/RealtimeEffectPanel.cpp

```diff
@@ -495,8 +495,17 @@ namespace
                mEffectState->SetActive(mEnableButton->IsDown());
                if (mProject)
                {
-                  ProjectHistory::Get(*mProject).ModifyState(false);
-                  UndoManager::Get(*mProject).MarkUnsaved();
+                  auto const effectName{GetEffectName(*mEffectState)};
+                  ProjectHistory::Get(*mProject).PushState(
+                     /*! i18n-hint: undo history record
+                     first parameter - realtime effect name
+                     */
+                     XO("Change settings for effect %s").Format(effectName),
```

Comment:
I think we have to add the name of the track it belongs to. If we don't,
1. I tweak my reverb effect on track 1,
2. I tweak my compressor effect on track 2,
3. I undo

would undo my tweaking of the reverb effect of track 1, too.
@LWinterberg please comment.

Reply:
I mean, it should undo the last effect, shouldn't it? I don't understand why the name of the track matters here. Does the code as-is effect all instances of an effect? so if:

1.  I tweak the reverb on track 1
2.  I tweak the reverb on track 2
3.  I tweak a second reverb on track 2
4.  I undo

would all changes across all tracks and effects be undone?

Reply:
Yes.
Why track name matters: this is what the history mechanism bases its logic to consolidate actions. If you change a reverb setting on track one and tell the history "Change reverb param", and then change a reverb setting on track 2 and tell the history "Change reverb param", it'll get consolidated.
So without the track name, it's as you described. With the track name:
* tweak effect on track 1: state is modified (meaning if I close the project, I'm asked if I want to save).
* tweak effect with same name on track 1 again: nothing new happens.
* tweak effect with same name on track 2: new history item is created. *

which I think is better, but please confirm.

Reply:
Another note: a new history item will only get added when I close an effect's UI. Testing this on @embarc-gabriel 's branch, not sure it's the same on master. Seems strange to me. Maybe we should polish this, too, before totally freezing audacity 3.